store/tracekv/store.go

```diff
@@ -90,6 +91,25 @@ func (tkv *Store) ReverseIterator(start, end []byte) types.Iterator {
 	return tkv.iterator(start, end, false)
 }
 
+// GetAllKeysUsedInTrace reads through all traced operations and returns
+// a set of all the keys inside the trace operations
+func (tkv *Store) GetAllKeysUsedInTrace(buf bytes.Buffer) map[string]bool {
+	var traceOp traceOperation
+	var err error
+	keys := make(map[string]bool)
```

Comment:
nit: this introduces unnecessary overhead for each element. Could use `struct{}` instead of `bool`, or in Go 1.18 could use a generic set. cc @liamsi

Could potentially leave for a future PR, though it would be a breaking change to the public interface so not ideal.

Reply:
switched to generic sets